src/provider/non_hiding_zeromorph.rs

```diff
@@ -514,87 +514,30 @@ where
 
 #[cfg(test)]
 mod test {
-  use std::iter;
-
-  use ff::{Field, PrimeField, PrimeFieldBits};
+  use ff::{Field, PrimeField};
   use halo2curves::bn256::Bn256;
   use halo2curves::bn256::Fr as Scalar;
   use itertools::Itertools as _;
-  use pairing::MultiMillerLoop;
-  use rand::thread_rng;
   use rand_chacha::ChaCha20Rng;
   use rand_core::SeedableRng;
 
   use super::quotients;
+  use crate::provider::util::test_utils::{test_fail_bad_proof, test_from_seed};
+  use crate::provider::Bn256EngineZM;
   use crate::{
     provider::{
-      keccak::Keccak256Transcript,
-      non_hiding_kzg::{UVKZGPoly, UniversalKZGParam},
-      non_hiding_zeromorph::{
-        batched_lifted_degree_quotient, eval_and_quotient_scalars, trim, ZMEvaluation, ZMPCS,
-      },
-      traits::DlogGroup,
-      Bn256Engine,
+      non_hiding_kzg::UVKZGPoly,
+      non_hiding_zeromorph::{batched_lifted_degree_quotient, eval_and_quotient_scalars, ZMPCS},
     },
     spartan::polys::multilinear::MultilinearPolynomial,
-    traits::{Engine as NovaEngine, Group, TranscriptEngineTrait, TranscriptReprTrait},
   };
 
-  fn commit_open_verify_with<E: MultiMillerLoop, NE: NovaEngine<GE = E::G1, Scalar = E::Fr>>()
```

Comment:
this test is quite useful and should be preserved, as a test, in its generic form (see the associated `test_commit_open_verify` which constitutes a formal rust test for this generic function)

Reply:
Sorry, I misinterpreted what it did and thought it only tested proving and verifying. Adding it back;

Reply:
I was wrong, you were right, I misread your test utils 🤦